lib/presentation/widgets/date_time_field.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:datetime_picker_formfield_new/datetime_picker_formfield.dart';
+import 'package:flutter/material.dart';
+import 'package:vikunja_app/core/utils/constants.dart';
+
+class VikunjaDateTimeField extends StatelessWidget {
+  final String label;
+  final void Function(DateTime?)? onSaved;
+  final void Function(DateTime?)? onChanged;
+  final DateTime? initialValue;
+  final EdgeInsetsGeometry padding;
+  final Icon icon;
+
+  const VikunjaDateTimeField({
+    Key? key,
+    required this.label,
+    this.onSaved,
+    this.onChanged,
+    this.initialValue,
+    this.padding = const EdgeInsets.symmetric(vertical: 10.0),
+    this.icon = const Icon(Icons.date_range),
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return DateTimeField(
+      initialValue: initialValue == null || initialValue!.year <= 1
+          ? null
+          : initialValue!.toLocal(),
+      format: vDateFormatLong,
+      decoration: InputDecoration(
+        labelText: label,
+        border: InputBorder.none,
+        icon: icon,
+      ),
+      onSaved: onSaved,
+      onChanged: onChanged,
+      onShowPicker: (context, currentValue) {
+        return _showDatePicker(context, currentValue ?? DateTime.now());
+      },
+    );
+  }
+
+  Future<DateTime?> _showDatePicker(context, currentValue) async {
+    var selectedDate = await showDialog<DateTime>(
+      context: context,
+      builder: (_) => DatePickerDialog(
+        initialDate: currentValue.year <= 1 ? DateTime.now() : currentValue,
+        firstDate: DateTime.now(),
```

Comment:
Does that mean it won't be possible to set dates earlier than today?

Reply:
That might not be desirable. It definitely is an inconsistency with how the web frontend works.

Is the value required or can we remove it?